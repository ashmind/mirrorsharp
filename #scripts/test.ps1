Set-StrictMode -Version 2
$ErrorActionPreference = 'Stop'

Write-Output 'dotnet test'
dotnet test
if ($LastExitCode -ne 0) {
    throw "dotnet test exited with code $LastExitCode"
}

Write-Output 'npm test'
try {
    Push-Location 'WebAssets'
    npm test
    if ($LastExitCode -ne 0) {
        throw "npm test exited with code $LastExitCode"
    }
}
finally {
    Pop-Location
}