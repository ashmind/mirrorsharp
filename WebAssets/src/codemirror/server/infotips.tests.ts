import { TestDriver, timers } from '../../testing/test-driver-jest';
import { INFOTIP_EVENTHANDLER } from './infotips.test.data';

const mockHoverDependencies = (driver: TestDriver) => {
    const cmView = driver.getCodeMirrorView();
    cmView.posAtCoords = () => 0;
    cmView.coordsAtPos = () => ({ left: 0, right: 0, top: 0, bottom: 0 });
};

const waitForHover = async (driver: TestDriver) => {
    timers.advanceTimersByTime(1000); // actual is 750, but just in case
    await driver.completeBackgroundWork();
};

test('hover requests infotip', async () => {
    const driver = await TestDriver.new({ text: 'test' });
    mockHoverDependencies(driver);

    driver.domEvents.mousemove(
        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
        driver.getCodeMirrorView().contentDOM.firstChild!
    );
    await waitForHover(driver);

    expect(driver.socket.sent).toContain('I0');
});

test('hover applies received infotip', async () => {
    const driver = await TestDriver.new({ text: 'test' });
    mockHoverDependencies(driver);

    driver.domEvents.mousemove(
        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
        driver.getCodeMirrorView().contentDOM.firstChild!
    );
    await waitForHover(driver);
    driver.receive.infotip(INFOTIP_EVENTHANDLER);
    await driver.completeBackgroundWork();

    const tooltip = driver.getCodeMirrorView().dom.querySelector('.cm-tooltip');
    expect(tooltip).toBeTruthy();
    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
    expect(tooltip!.innerHTML).toMatchSnapshot();
});