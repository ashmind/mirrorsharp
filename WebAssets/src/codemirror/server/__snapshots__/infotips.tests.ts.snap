// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hover applies received infotip 1`] = `"<div class="mirrorsharp-infotip cm-tooltip-section"><div class="mirrorsharp-parts-section"><span class="mirrorsharp-infotip-icon mirrorsharp-infotip-icon-delegate mirrorsharp-infotip-icon-public"></span><span class="tok-keyword">delegate</span><span class="tok-space"> </span><span class="tok-keyword">void</span><span class="tok-space"> </span><span class="tok-namespace">System</span><span class="tok-punctuation">.</span><span class="tok-delegate">EventHandler</span><span class="tok-punctuation">(</span><span class="tok-keyword">object</span><span class="tok-space"> </span><span class="tok-parameter">sender</span><span class="tok-punctuation">,</span><span class="tok-space"> </span><span class="tok-namespace">System</span><span class="tok-punctuation">.</span><span class="tok-class">EventArgs</span><span class="tok-space"> </span><span class="tok-parameter">e</span><span class="tok-punctuation">)</span></div><div class="mirrorsharp-parts-section"><span class="tok-text">Represents the method that will handle an event that has no event data.</span></div></div>"`;
